src/BeadingStrategy/RedistributeBeadingStrategy.cpp

```diff
@@ -37,7 +37,11 @@ coord_t RedistributeBeadingStrategy::getTransitionThickness(coord_t lower_bead_c
 
 coord_t RedistributeBeadingStrategy::getOptimalBeadCount(coord_t thickness) const
 {
-    return parent->getOptimalBeadCount(thickness);
+    if (thickness <= 2 * optimal_width_outer)
+    {
+        return thickness / 2 >= minimum_variable_line_width * optimal_width_outer ? 2 : 1;
```

Comment:
> but I favor multiplication over division as the former is more performant.

I favor readability over such small concerns. If we're looking for performance, after all, I should have written `thickness >> 1`, as bitshift is more performant than either multiplication or division ;-)

What I mean by readability is that right now, it self-documents as 'half the thickness is >= (something)'. We're trying to outsmart the compiler for something we _know_ it can optimize. (Integer division by two.)

Reply:
Speaking of readability.....

> Edit: ah I think i get it. the minimum_variable_line_width is less then 1 and describes the fractional width you can print of the optimal_width_outer?

Yes, that's it. Perhaps we should change the name (and type) to ratio, since I can see that this will confuse the heck out of people the way it's now. I didn't change the name since it already existed, but can be added to this PR as well. (Or done later.)

Reply:
I've now renamed the variable.